File: examples/docs-code-snippets/src/main/java/examples/java/recipes/RecipeWithFailureStrategy.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.InteractionDescriptor;
import com.ing.baker.recipe.javadsl.InteractionFailureStrategy;
import com.ing.baker.recipe.javadsl.Recipe;
import examples.java.interactions.ShipOrder;

public class RecipeWithFailureStrategy {

    public final static Recipe recipe = new Recipe("example")
        .withInteractions(
            InteractionDescriptor.of(ShipOrder.class)
                .withFailureStrategy(
                        InteractionFailureStrategy.FireEvent("shippingFailed")
                )
        );
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithFailureStrategy.scala
package examples.scala.recipes

import com.ing.baker.recipe.common.InteractionFailureStrategy
import com.ing.baker.recipe.scaladsl.Recipe
import examples.scala.interactions.ShipOrder

object RecipeWithFailureStrategy {
  val recipe: Recipe = Recipe("example")
    .withInteraction(
      ShipOrder.interaction
        .withFailureStrategy(
          InteractionFailureStrategy.FireEventAfterFailure(Some("shippingFailed"))
        )
    )
}